scipy/stats/_stats_py.py

```diff
@@ -37,7 +37,12 @@
 from numpy.testing import suppress_warnings
 
 from scipy.spatial.distance import cdist
+from scipy.spatial import distance_matrix
+from scipy.sparse import hstack, vstack, coo_array, block_diag, eye
```

Comment:
```suggestion
from scipy.sparse import coo_array, block_diag
```
It's OK with me to import these, but in the code below, it would not be immediately obvious whether `hstack`, etc. are for NumPy arrays or sparse arrays. Personally, I'd suggest:
```
from scipy import sparse
```
and always using fully-qualified names.